app/helpers/application_helper.rb

```diff
@@ -8,14 +8,21 @@ def title(text)
     text
   end
 
-  RENDERER = Redcarpet::Markdown.new(Redcarpet::Render::HTML.new)
   def render_markdown(text)
-    RENDERER.render text
+    # Using UNSAGE flag here because we have further processing below
+    CommonMarker.render_html text, %i[UNSAFE], %i[autolink tagfilter]
   end
 
   def safe_render_markdown(text, options = {})
+    # Markdown pipeline :)
     scrubber = options[:scrubber] || PostScrubber.new
-    raw(sanitize(render_markdown(text), scrubber: scrubber))
+    s = sanitize(render_markdown(text), scrubber: scrubber)
+    doc = Nokogiri::HTML::DocumentFragment.parse s
+    doc.css('a[href]').each do |a|
+      # Like Stack Exchange does
+      a['rel'] = 'nofollow noreferrer'
+    end
```

Comment:
The Nokogiri parse is unnecessary here: `noopener noreferrer` only applies to links that also have `target="_blank"`, i.e. open in a new tab. Since Markdown links don't, you don't need to add the `rel` and thus can avoid having to parse the document (so, yay, extra performance gain).

Reply:
It's not `noopener`, but `nofollow`, which is in line with that on SE. Though you're right it might be better to remove this patch altogether.